python/setup.py

```diff
@@ -16,26 +17,77 @@
 from setuptools import setup, find_packages
 from setuptools.command.build_py import build_py
 from setuptools.command.develop import develop
+from setuptools.command.install import install
 from setuptools.command.test import test as test_command
 from typing import List
+from wheel.bdist_wheel import bdist_wheel as _bdist_wheel
 
 PLATFORM = 'unix'
 if platform.platform().startswith('Win'):
     PLATFORM = 'win'
 
 MODEL_DIR = os.path.join('stan', PLATFORM)
 MODEL_TARGET_DIR = os.path.join('prophet', 'stan_model')
-
+CMDSTAN_VERSION = "2.26.1"
 
 def get_backends_from_env() -> List[str]:
-    from prophet.models import StanBackendEnum
-    return os.environ.get("STAN_BACKEND", StanBackendEnum.PYSTAN.name).split(",")
-
+    return os.environ.get("STAN_BACKEND", "PYSTAN").split(",")
+
+def prune_cmdstan_files(cmdstan_dir):
+    """
+    Remove unnecessary folders from the unbundled cmdstan installation.
+    We only need to keep the files that will be used to execute the model binary at runtime.
+    """
+    remove_dirs = ["stan"]
+    for dirname in remove_dirs:
+        rmtree(os.path.join(cmdstan_dir, dirname))
+    raw_binaries = ["linux-stanc", "mac-stanc", "windows-stanc"]  # These are converted into "stanc" in the Make process
+    for fname in raw_binaries:
+        os.remove(os.path.join(cmdstan_dir, f"bin/{fname}"))
+
+def build_cmdstan_model(target_dir):
+    import cmdstanpy
+    cmdstanpy.install_cmdstan(version=CMDSTAN_VERSION, dir=target_dir, overwrite=True)
+    cmdstan_dir = os.path.join(target_dir, f"cmdstan-{CMDSTAN_VERSION}")
+    cmdstanpy.set_cmdstan_path(cmdstan_dir)
+    model_name = 'prophet.stan'
+    target_name = 'prophet_model.bin'
+    sm = cmdstanpy.CmdStanModel(stan_file=os.path.join(MODEL_DIR, model_name))
+    sm.compile()
+    copy(sm.exe_file, os.path.join(target_dir, target_name))
+    prune_cmdstan_files(cmdstan_dir)
+
+def build_pystan_model(target_dir):
+    import pystan
+    model_name = 'prophet.stan'
+    target_name = 'prophet_model.pkl'
+    with open(os.path.join(MODEL_DIR, model_name)) as f:
+        model_code = f.read()
+    sm = pystan.StanModel(model_code=model_code)
+    with open(os.path.join(target_dir, target_name), 'wb') as f:
+        pickle.dump(sm, f, protocol=pickle.HIGHEST_PROTOCOL)
 
 def build_models(target_dir):
-    from prophet.models import StanBackendEnum
     for backend in get_backends_from_env():
-        StanBackendEnum.get_backend_class(backend).build_model(target_dir, MODEL_DIR)
+        print(f"Compiling {backend} model")
+        if backend == "CMDSTANPY":
+            build_cmdstan_model(target_dir)
+        elif backend == "PYSTAN":
+            build_pystan_model(target_dir)
+
+
+class BDistWheelNonPure(_bdist_wheel):
```

Comment:
This ensures that the wheel gets tagged with the appropriate operating system / platform, not just "any".